optuna/visualization/_contour.py

```diff
@@ -137,7 +137,7 @@ def _get_contour_plot(
             min_value = math.pow(10, math.log10(min_value) - padding)
             max_value = math.pow(10, math.log10(max_value) + padding)
 
-        elif _is_categorical(trials, p_name):
+        elif not _is_numerical(trials, p_name):
```

Comment:
[nits]

Can we rewrite here like following? (if it's not, please leave it as is)

```python
        elif _is_numerical(trials, p_name):
            padding = (max_value - min_value) * padding_ratio
            min_value = min_value - padding
            max_value = max_value + padding

        else:
            # Plotly>=4.12.0 draws contours using the indices of categorical variables instead of
            # raw values and the range should be updated based on the cardinality of categorical
            # variables. See https://github.com/optuna/optuna/issues/1967.
            if version.parse(plotly.__version__) >= version.parse("4.12.0"):
                span = len(set(values)) - 1
                padding = span * padding_ratio
                min_value = -padding
                max_value = span + padding
```

Reply:
Thanks! The modification should be included in this pull request, 1d62f3b.